extractor/src/test/java/org/schabi/newpipe/extractor/services/soundcloud/SoundcloudPlaylistExtractorTest.java

```diff
@@ -229,6 +234,15 @@ public static void setUp() throws Exception {
         public void testGetPageInNewExtractor() throws Exception {
             final PlaylistExtractor newExtractor = SoundCloud.getPlaylistExtractor(extractor.getUrl());
             defaultTestGetPageInNewExtractor(extractor, newExtractor);
+            String page1 = newExtractor.getNextPageUrl();
+            defaultTestMoreItems(newExtractor); // there has to be another page
+            String page2 = newExtractor.getNextPageUrl();
+            defaultTestMoreItems(newExtractor); // and another one
+            String page3 = newExtractor.getNextPageUrl();
+
+            assertNotEquals("Same pages", page1, page2);
+            assertNotEquals("Same pages", page2, page3);
+            assertNotEquals("Same pages", page3, page1);
```

Comment:
What are you even doing here? getNextPageUrl should always return the same URL, as it's the next page relative to the initial page, see https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/ListExtractor.java#L36.

Reply:
So why did the old extractor do the same thing as I do? Every time getPage was called it overwrote nextPageUrl with the url of the page coming after the one provided as the argument of getPage, thus having the same behaviour as now.

Reply:
I don't know, that would be a bug. YouTube doesn't do it: https://github.com/TeamNewPipe/NewPipeExtractor/blob/dev/extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubePlaylistExtractor.java#L81, and YouTube is the 'golden' service implementation.

Reply:
Ok, will change this ;-)